lib/ex_unit/lib/ex_unit/doc_test.ex

```diff
@@ -509,84 +514,96 @@ defmodule ExUnit.DocTest do
+    do_extract_tests(lines, expr_acc <> "\n" <> string, expected_acc, acc, newtest, module)
   end
 
   # Expression numbers are simply skipped.
-  defp extract_tests([{<<"iex(", _>> <> string, line} | lines], expr_acc, expected_acc, acc, newtest) do
-    extract_tests([{"iex" <> skip_iex_number(string), line} | lines], expr_acc, expected_acc, acc, newtest)
+  defp do_extract_tests([{<<"iex(", _>> <> string = line, line_no} | lines],
+      expr_acc, expected_acc, acc, newtest, module) do
+    do_extract_tests([{"iex" <> skip_iex_number(string, module, line_no, line), line_no} | lines],
+      expr_acc, expected_acc, acc, newtest, module)
   end
 
   # Expression numbers are simply skipped redux.
-  defp extract_tests([{<<"...(", _>> <> string, line} | lines], expr_acc, expected_acc, acc, newtest) do
-    extract_tests([{"..." <> skip_iex_number(string), line} | lines], expr_acc, expected_acc, acc, newtest)
+  defp do_extract_tests([{<<"...(", _>> <> string, line_no} = line | lines],
+      expr_acc, expected_acc, acc, newtest, module) do
+    do_extract_tests([{"..." <> skip_iex_number(string, module, line_no, line), line_no} | lines],
+      expr_acc, expected_acc, acc, newtest, module)
   end
 
   # Skip empty or documentation line.
-  defp extract_tests([_ | lines], "", "", acc, _) do
-    extract_tests(lines, "", "", acc, true)
+  defp do_extract_tests([_ | lines], "", "", acc, _, module) do
+    do_extract_tests(lines, "", "", acc, true, module)
   end
 
   # Encountered an empty line, store pending test
-  defp extract_tests([{"", _} | lines], expr_acc, expected_acc, [test | t], _) do
+  defp do_extract_tests([{"", _} | lines], expr_acc, expected_acc, [test | t], _, module) do
     test = add_expr(test, expr_acc, expected_acc)
-    extract_tests(lines, "", "", [test | t], true)
+    do_extract_tests(lines, "", "", [test | t], true, module)
   end
 
   # Finally, parse expected_acc.
-  defp extract_tests([{expected, _} | lines], expr_acc, "", acc, newtest) do
-    extract_tests(lines, expr_acc, expected, acc, newtest)
+  defp do_extract_tests([{expected, _} | lines], expr_acc, "", acc, newtest, module) do
+    do_extract_tests(lines, expr_acc, expected, acc, newtest, module)
   end
 
-  defp extract_tests([{expected, _} | lines], expr_acc, expected_acc, acc, newtest) do
-    extract_tests(lines, expr_acc, expected_acc <> "\n" <> expected, acc, newtest)
+  defp do_extract_tests([{expected, _} | lines], expr_acc, expected_acc, acc, newtest, module) do
+    do_extract_tests(lines, expr_acc, expected_acc <> "\n" <> expected, acc, newtest, module)
   end
 
-  defp skip_iex_number(")>" <> string) do
+  defp skip_iex_number(")>" <> string, _module, _line_no, _line) do
     ">" <> string
   end
 
-  defp skip_iex_number(<<_>> <> string) do
-    skip_iex_number(string)
+  defp skip_iex_number("", module, line_no, line) do
+    raise Error, line: line_no, module: module,
+                 message: ~s{unknown IEx prompt: "#{line}".\n} <>
```

Comment:
just to double check for future cases. we know for sure line is a string, because it comes from String.split/ we still use `inspect`, right?